python/src/integry/resources/functions/types.py

```diff
@@ -124,6 +124,45 @@ def get_langchain_tool[
         )
         return tool
 
+    def get_llamaindex_tool[
+        T
+    ](
+        self, tool_from_defaults: Callable[..., T], tools_metadata: Callable[..., T], user_id: str) -> T:
+        
+        """
+        Returns a llamaIndex tool for the function..
+
+        Args:
+            tool_from_defaults: This should be llamaIndex `FunctionTool.from_defaults` method.
+            tools_metadata: LlamaIndex function that generates metadata for the tool.
+            user_id: The user ID for authentication.
+
+        Returns:
+            llamaindex tool
+        """
+
+        async def execute_function(**kwargs: Dict[str, Any]) -> FunctionCallOutput:
+            callable_function = self._get_callable(user_id=user_id)
+            result = await callable_function(**kwargs)
+            return result
+
+        function_schema = get_pydantic_model_from_json_schema(
+            json_schema=self.get_json_schema()['parameters'],
+        )
+
+        metadata = tools_metadata(
+            name=self.name,
+            description=self.description,
+            fn_schema=function_schema,
+        )
+
+        async_fn: Callable[[Dict[str, Any]], Awaitable[Any]] = lambda **kwargs: execute_function(**kwargs)
+
+        return tool_from_defaults(
+            async_fn = async_fn,
+            tool_metadata=metadata,
+        )
```

Comment:
Since we are just calling `_get_callable`, can we do this instead

```suggestion
        function_schema = get_pydantic_model_from_json_schema(
            json_schema=self.get_json_schema()['parameters'],
        )

        metadata = tools_metadata(
            name=self.name,
            description=self.description,
            fn_schema=function_schema,
        )

        return tool_from_defaults(
            async_fn=self._get_callable(user_id=user_id),
            tool_metadata=metadata,
        )
```